JavaScript/Topics/5.Syntax/6.Functions/Video.md

```diff
@@ -0,0 +1,122 @@
+
+
+## Javascript Functions
+
+<iframe  width="560"  height="315"  src="https://www.youtube.com/embed/LAl0vdiPVZQ"  title="YouTube video player"  frameborder="0"  allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"  allowfullscreen></iframe>
+
+
+## Transcript
+
+A **JavaScript function** is a block of code designed to perform a particular task. Its basically dividing you program into different parts.
+
+Now a function looks like this..  
+Wait.. lets break it into different parts first
+
+```javascript
+function myFunction(p1, p2) {
+    return p1 * p2; // The function returns the product of p1 and p2
+}
+
+But before learning functions, lets learn about **scope**.
+
+<hr>
+
+### Scope
+
+The scope is basically the availability of your code.  
+Let’s see by an example
+
+``` javascript
+if (true) {
+    const a = 2;
+}
+
+console.log(a); //ReferenceError
+
+This time javascript throws an error. Why is it so?
```

Comment:
```suggestion
```javascript
function myFunction(p1, p2) {
    return p1 * p2; // The function returns the product of p1 and p2
}

But before learning functions, lets learn about **scope**.

<hr>

### Scope

The scope is basically the availability of your code.  
Let’s see by an example

``` javascript
if (true) {
    const a = 2;
}

console.log(a); //ReferenceError
```

This time javascript throws an error. Why is it so?
```

You forgot to turn off the code-block.